presto-main/src/main/java/com/facebook/presto/execution/QueryManagerConfig.java

```diff
@@ -45,6 +45,8 @@
     private Duration remoteTaskMinErrorDuration = new Duration(2, TimeUnit.MINUTES);
     private int remoteTaskMaxCallbackThreads = 1000;
 
+    private Duration queryMaxRuntime = new Duration(1, TimeUnit.HOURS);
```

Comment:
This is too short for a default. We have many queries that run for several hours and that is not an unreasonable thing to do. You would also be upset to discover this new default when your query fails after running for an hour.

I think one day would be a better default. (you could run longer than that but in our experience any query older than a day is a bug in something)